File: test/api_test.js
const https = require('https');

const baseUrl = 'https://sochio-backend.onrender.com';
const apiSecret = 'sochio-api-secret-2024';

async function makeRequestWithAuth(method, endpoint, data = null, token = null) {
  return new Promise((resolve, reject) => {
    const url = new URL(endpoint, baseUrl);
    const options = {
      method,
      headers: {
        'Authorization': `Bearer ${apiSecret}`,
        'Content-Type': 'application/json'
      }
    };
    
    if (token) {
      options.headers['key'] = token;
    }

    const req = https.request(url, options, (res) => {
      let body = '';
      res.on('data', chunk => body += chunk);
      res.on('end', () => {
        resolve({
          statusCode: res.statusCode,
          body: body,
          headers: res.headers
        });
      });
    });

    req.on('error', reject);
    
    if (data) {
      req.write(JSON.stringify(data));
    }
    
    req.end();
  });
}

async function makeRequest(method, endpoint, data = null) {
  return new Promise((resolve, reject) => {
    const url = new URL(endpoint, baseUrl);
    const options = {
      method,
      headers: {
        'Authorization': `Bearer ${apiSecret}`,
        'Content-Type': 'application/json'
      }
    };

    const req = https.request(url, options, (res) => {
      let body = '';
      res.on('data', chunk => body += chunk);
      res.on('end', () => {
        resolve({
          statusCode: res.statusCode,
          body: body,
          headers: res.headers
        });
      });
    });

    req.on('error', reject);
    
    if (data) {
      req.write(JSON.stringify(data));
    }
    
    req.end();
  });
}

async function testHealthCheck() {
  console.log('📡 Testing Health Check...');
  try {
    const response = await makeRequest('GET', '/api/health');
    console.log(`Status: ${response.statusCode}`);
    console.log(`Response: ${response.body}`);
    
    if (response.statusCode === 200) {
      console.log('✅ Health check passed\n');
    } else {
      console.log('❌ Health check failed\n');
    }
  } catch (error) {
    console.log(`❌ Health check error: ${error.message}\n`);
  }
}

async function testLogin() {
  console.log('🔐 Testing Login...');
  try {
    const loginData = {
      email: 'admin@sochio.com',
      password: 'Admin@123'
    };
    
    const response = await makeRequest('POST', '/admin/login', loginData);
    console.log(`Status: ${response.statusCode}`);
    console.log(`Response: ${response.body}`);
    
    if (response.statusCode === 200) {
      const data = JSON.parse(response.body);
      console.log('✅ Login successful');
      console.log(`Token: ${data.token?.substring(0, 20)}...\n`);
      return data.token;
    } else {
      console.log('❌ Login failed\n');
      return null;
    }
  } catch (error) {
    console.log(`❌ Login error: ${error.message}\n`);
    return null;
  }
}

async function testGetProducts(token) {
  console.log('📦 Testing Get Products...');
  try {
    const options = {
      method: 'GET',
      headers: {
        'Authorization': `Bearer ${apiSecret}`,
        'Content-Type': 'application/json'
      }
    };
    
    if (token) {
      options.headers['key'] = token;
    }
    
    const response = await makeRequestWithAuth('GET', '/product', null, token);
    console.log(`Status: ${response.statusCode}`);
    console.log(`Response: ${response.body}`);
    
    if (response.statusCode === 200) {
      const data = JSON.parse(response.body);
      console.log('✅ Products retrieved successfully');
      console.log(`Product count: ${data.product?.length || 0}\n`);
    } else {
      console.log('❌ Get products failed\n');
    }
  } catch (error) {
    console.log(`❌ Get products error: ${error.message}\n`);
  }
}

async function testGetCategories(token) {
  console.log('📂 Testing Get Categories...');
  try {
    const response = await makeRequestWithAuth('GET', '/category', null, token);
    console.log(`Status: ${response.statusCode}`);
    console.log(`Response: ${response.body}`);
    
    if (response.statusCode === 200) {
      const data = JSON.parse(response.body);
      console.log('✅ Categories retrieved successfully');
      console.log(`Category count: ${data.category?.length || 0}\n`);
    } else {
      console.log('❌ Get categories failed\n');
    }
  } catch (error) {
    console.log(`❌ Get categories error: ${error.message}\n`);
  }
}

async function runTests() {
  console.log('🚀 Starting API Endpoint Tests...\n');
  
  try {
    await testHealthCheck();
    const token = await testLogin();
    await testGetProducts(token);
    await testGetCategories(token);
  } catch (error) {
    console.log(`❌ Test suite failed: ${error.message}`);
  }
  
  console.log('✅ API Tests Completed');
}

runTests();

File: test/api_test.dart
import 'dart:convert';
import 'dart:io';

const String baseUrl = 'https://sochio-backend.onrender.com';
const String apiSecret = 'sochio-api-secret-2024';

void main() async {
  print('🚀 Starting API Endpoint Tests...\n');
  
  final client = HttpClient();
  
  try {
    // Test 1: Health Check
    await testHealthCheck(client);
    
    // Test 2: Login
    String? token = await testLogin(client);
    
    // Test 3: Get Videos
    await testGetVideos(client, token);
    
    // Test 4: Payment Creation
    await testPaymentCreation(client, token);
    
  } catch (e) {
    print('❌ Test suite failed: $e');
  } finally {
    client.close();
    print('\n✅ API Tests Completed');
  }
}

Future<void> testHealthCheck(HttpClient client) async {
  print('📡 Testing Health Check...');
  try {
    final request = await client.getUrl(Uri.parse('$baseUrl/api/health'));
    request.headers.set('Authorization', 'Bearer $apiSecret');
    
    final response = await request.close();
    final responseBody = await response.transform(utf8.decoder).join();
    
    print('Status: ${response.statusCode}');
    print('Response: $responseBody');
    
    if (response.statusCode == 200) {
      print('✅ Health check passed\n');
    } else {
      print('❌ Health check failed\n');
    }
  } catch (e) {
    print('❌ Health check error: $e\n');
  }
}

Future<String?> testLogin(HttpClient client) async {
  print('🔐 Testing Login...');
  try {
    final request = await client.postUrl(Uri.parse('$baseUrl/api/auth/login'));
    request.headers.set('Content-Type', 'application/json');
    request.headers.set('Authorization', 'Bearer $apiSecret');
    
    final loginData = {
      'email': 'admin@sochio.com',
      'password': 'admin123'
    };
    
    request.write(jsonEncode(loginData));
    
    final response = await request.close();
    final responseBody = await response.transform(utf8.decoder).join();
    
    print('Status: ${response.statusCode}');
    print('Response: $responseBody');
    
    if (response.statusCode == 200) {
      final data = jsonDecode(responseBody);
      final token = data['token'];
      print('✅ Login successful');
      print('Token: ${token?.substring(0, 20)}...\n');
      return token;
    } else {
      print('❌ Login failed\n');
      return null;
    }
  } catch (e) {
    print('❌ Login error: $e\n');
    return null;
  }
}

Future<void> testGetVideos(HttpClient client, String? token) async {
  print('🎥 Testing Get Videos...');
  try {
    final request = await client.getUrl(Uri.parse('$baseUrl/api/videos'));
    request.headers.set('Authorization', 'Bearer $apiSecret');
    if (token != null) {
      request.headers.set('x-auth-token', token);
    }
    
    final response = await request.close();
    final responseBody = await response.transform(utf8.decoder).join();
    
    print('Status: ${response.statusCode}');
    print('Response: $responseBody');
    
    if (response.statusCode == 200) {
      final data = jsonDecode(responseBody);
      print('✅ Videos retrieved successfully');
      print('Video count: ${data['videos']?.length ?? 0}\n');
    } else {
      print('❌ Get videos failed\n');
    }
  } catch (e) {
    print('❌ Get videos error: $e\n');
  }
}

Future<void> testPaymentCreation(HttpClient client, String? token) async {
  print('💳 Testing Payment Creation...');
  try {
    final request = await client.postUrl(Uri.parse('$baseUrl/api/payments/create'));
    request.headers.set('Content-Type', 'application/json');
    request.headers.set('Authorization', 'Bearer $apiSecret');
    if (token != null) {
      request.headers.set('x-auth-token', token);
    }
    
    final paymentData = {
      'amount': 1000,
      'currency': 'INR',
      'productId': 'test_product_123',
      'userId': 'test_user_456'
    };
    
    request.write(jsonEncode(paymentData));
    
    final response = await request.close();
    final responseBody = await response.transform(utf8.decoder).join();
    
    print('Status: ${response.statusCode}');
    print('Response: $responseBody');
    
    if (response.statusCode == 200 || response.statusCode == 201) {
      print('✅ Payment creation successful\n');
    } else {
      print('❌ Payment creation failed\n');
    }
  } catch (e) {
    print('❌ Payment creation error: $e\n');
  }
}